### DISCUSSION TITLE
Does walletconnect V2.0 have a method to check whether the network is normal?

### DISCUSSION THREAD

**Discussion by xtjcmavg** [WalletConnect/walletconnect-monorepo#2796] [Developer Support] [answered]
After the wallet APP is successfully connected to the DAPP, perform the following test operations:
Close or minimize the DAPP in the browser, and then disconnect the wallet APP from the DAPP network.
Returning to DAPP again, operations such as querying the balance cannot be performed normally.
In this case, how should we judge whether the connection exists?
No method found in the UniversalProvider object that can check whether the network is healthy

**Comment by crypblizz8** [comment, marked as answer]
Hey @xtjcmavg, you could listen to events here on your dapp side to see if connections were disconnected:
https://docs.walletconnect.com/2.0/web/providers/universal#events
